### DISCUSSION TITLE
EEPROM Problem

### DISCUSSION THREAD

**Discussion by kuckyder2te** [earlephilhower/arduino-pico#657] [Q&A] [answered]
Hello, everyone,
I had already posted the problem in the SDK forum.
"EEPROM Lib from "earlephilhower". If i compile the example "eeprom_write" i get this message
"lib\EEPROM\EEPROM.cpp:121:5: error: 'rp2040' was not declared in this scope"
in line
" rp2040.idleOtherCore();"
"
I program via PlatformIO and C++. I'm not sure if I understood "earlephilhower" correctly. I select the RaspberryPi Pico Board. Do I need to install additional software packages? I thought they were already there.
Greetings Willi

**Comment by earlephilhower** [comment, marked as answer]
It looks like you manually copied files from this core to you project which is running the Arduino.cc MBED core.

That is not going to work.  You do not need to copy any files to make it work, but you do need to use this core and not the MBED one.  See https://github.com/platformio/platform-raspberrypi/pull/36 for instructions on installing this code (you'll also need to delete any files you manually coped from here to you project, too).

**Reply by kuckyder2te** [reply]
I understand, thanks

**Reply by kuckyder2te** [reply]
After many attempts, I finally managed to compile the EEPROM sketch. Now I have the problem that the sketch is no longer flashed. The dots appear and I press reset and run, but the process is aborted.
I'm an "experienced" beginner, but I can't get any further here.
I'm working with VSCode via PlatformIO.
I'm grateful for any help.
willy
PS I read Samuel Pinsch's block on this topic.

**Reply by kuckyder2te** [reply]
This is my PlatformIO.ini
[env:pico]
;platform = raspberrypi
platform = https://github.com/maxgerhardt/platform-raspberrypi.git
board_build.core = earlephilhower
board = pico
framework = arduino
;upload_port = COM17
monitor_port = COM20
lib_deps = 
	martinsos/HCSR04@^2.0.0
	bakercp/Logger@^1.0.3
	hideakitai/TaskManager@^0.4.8
	nrf24/RF24@^1.4.2
	adafruit/Adafruit BMP280 Library@^2.6.2
	asukiaaa/MPU9250_asukiaaa@^1.5.11
	; milesburton/DallasTemperature@^3.9.1
	mike-matera/FastPID@^1.3.1
	khoih-prog/RP2040_PWM@^1.3.0
;	paolop74/extEEPROM@^3.4.1
build_flags = 
	-DDEBUG=VERBOSE
;	-DDEBUG=NOTICE
;	-DDEBUG=WARNING
;	-DDEBUG=FATAL
;	-D_PID_ADJUST
;	-D_MINITERM

**Reply by earlephilhower** [reply]
Check #803 for upload issues.  Something may have gotten borked w/the P.IO default upload method.

**Reply by kuckyder2te** [reply]
Dear earlephilhower,
thanks for the quick response. I will try it on weekend. 
I'm really very grateful.
Willy

**Reply by kuckyder2te** [reply]
The problem is solved.
But I always have to press the BOOTSEL button and reset it.
Then I need to copy the "firmware.elf.uf2" file to the pico drive.
done

Thanks to all
